nixos/modules/services/web-apps/ixp-manager.nix

```diff
@@ -0,0 +1,521 @@
+        IXP_RPKI_RTR1_PORT = "3323";
+      };
+    };
+
+    poolConfig = mkOption {
+      type = with types; attrsOf (oneOf [ str int bool ]);
+      default = {
+        "pm" = "dynamic";
+        "pm.max_children" = 32;
+        "pm.start_servers" = 2;
+        "pm.min_spare_servers" = 2;
+        "pm.max_spare_servers" = 4;
+        "pm.max_requests" = 500;
+      };
+      description = lib.mdDoc ''
+        Options for the IXP-Manager PHP pool. See the documentation on `php-fpm.conf`
+        for details on configuration directives.
+      '';
+    };
+
+    nginx = mkOption {
+      type = types.submodule (
+        recursiveUpdate
+          (import ../web-servers/nginx/vhost-options.nix { inherit config lib; }) { }
+      );
+      default = { };
+      example = literalExpression ''
+        {
+          serverAliases = [
+            "ixp.''${config.networking.domain}"
+          ];
+          # To enable encryption and let let's encrypt take care of certificate
+          forceSSL = true;
+          enableACME = true;
+          # To set the IXP-Manager virtualHost as the default virtualHost
+          default = true;
+        }
+      '';
+      description = mdDoc ''
+        With this option, you can customize the nginx virtualHost settings.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = cfg.createDatabaseLocally -> cfg.settings.DB_HOST == "localhost";
+        message = "The database host must be \"localhost\" if services.ixp-manager.createDatabaseLocally is set to true.";
+      }
+    ];
+
+    services.ixp-manager.settings = mkIf cfg.enableMRTG {
+      GRAPHER_BACKEND_MRTG_DBTYPE = "rrd";
+      GRAPHER_BACKEND_MRTG_WORKDIR = "/var/lib/mrtg";
+      GRAPHER_BACKEND_MRTG_LOGDIR = "/var/lib/mrtg";
+    };
+
+    users.users.${cfg.user} = {
+      group = "${cfg.group}";
+      isSystemUser = true;
+    };
+
+    users.groups.${cfg.group} = { };
+
+    environment.systemPackages = [ artisanWrapper ];
+
+    services.nginx = {
+      enable = true;
+      virtualHosts."${cfg.hostname}" = mkMerge [
+        cfg.nginx
+        {
+          root = mkForce "${package}/public";
+          locations."/" = {
+            index = "index.php";
+            tryFiles = "$uri $uri/ /index.php?$query_string";
+          };
+          locations."~ .php$".extraConfig = ''
+            fastcgi_pass unix:${config.services.phpfpm.pools."ixp-manager".socket};
+            fastcgi_split_path_info ^(.+\.php)(/.+)$;
+          '';
+        }
+      ];
+    };
+
+    services.phpfpm.pools.ixp-manager = {
+      user = cfg.user;
+      group = cfg.group;
+      phpPackage = phpPackage;
+      settings = {
+        "listen.mode" = "0660";
+        "listen.owner" = config.services.nginx.user;
+        "listen.group" = config.services.nginx.group;
+      } // cfg.poolConfig;
+    };
+
+    services.mysql = mkIf cfg.createDatabaseLocally {
+      enable = true;
+      package = pkgs.mysql80;
```

Comment:
is it smart pinning the mysql version here ?

Reply:
there is unfortunately no other way. ixp-manager only works with mysql 8...